src/charmlibs/pathops/__init__.py

```diff
@@ -12,7 +12,43 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-"""Pathlib-like interface for local and ops.Container filesystem paths."""
+r""":mod:`pathlib`-like interface for local and :class:`ops.Container` filesystem paths.
+
+:class:`ContainerPath` implements a ``pathlib.Path`` style interface for path operations on
+a Juju charm's workload container.
+This interface is defined in :class:`PathProtocol`, which defines a subset of ``pathlib.Path``
+methods that can be implemented via Pebble. The file creation methods are extended with ownership
```

Comment:
Suggest rewording to avoid "via":

```suggestion
methods that Pebble supports. The file creation methods are extended with ownership
```